src/storage/sql_table.cpp

```diff
@@ -6,11 +6,11 @@
 
 namespace terrier::storage {
 
-SqlTable::SqlTable(BlockStore *const store, const catalog::Schema &schema, const catalog::table_oid_t oid)
-    : block_store_(store), oid_(oid) {
+SqlTable::SqlTable(BlockStore *const store, catalog::Schema schema, const catalog::table_oid_t oid)
+    : block_store_(store), oid_(oid), schema_(std::move(schema)) {
```

Comment:
This is a very significant API change.  If you're going to pass it in what is the justification for changing it from a reference to a move rather than a copy?

Reply:
This change is prompted by clang-tidy. It thinks, for c++17, since we have std::move, passing a value and then move it is better than passing a reference and then copy it inside the function body. For callers, this change has no impact. They do not need to change anything.

Reply:
We updated this constructor according to the latest master and clang-tidy stopped prompting this change. So we reverted this API change.